update SNV methods (#61)
AlexsLemonade/OpenPBTA-manuscript: content/03.methods.md

#### SNV and INDEL calling

We used four variant callers to call SNVs and INDELS from targeted DNA panel, WXS, and WGS data: Strelka2, Mutect2, Lancet, and VarDict.

> The input interval BED files for both panel and WXS data, derived from probes used to create the assays, were padded by 100 bp on each side for all variant calling algorithm runs.

Reviewer: ```suggestion
The input interval BED files for both panel and WXS data were derived from probes used to create the assays,  padded by 100 bp on each side for all variant calling algorithm runs.
```

> We utilized the non-padded BROAD Institute interval calling list `wgs_calling_regions.hg38.interval_list`, comprised of the full genome minus N bases, as the WGS input BED, unless otherwise noted below. 

Reviewer: ```suggestion
For WGS calling, we used the non-padded BROAD Institute interval calling list `wgs_calling_regions.hg38.interval_list`, comprised of the full genome minus N bases, as the WGS input BED, unless otherwise noted below. 
```

Is there a link to this file we can add?

Author: updated in upcoming commit

> Strelka2 [@doi:10/gdwrp4] v2.9.3 was run using default parameters on human genome reference hg38, canonical chromosomes only (chr1-22, X,Y,M), as recommended by the authors.

Reviewer: Just to clarify, this is also using the BROAD intervals? Having this sentence after the discussion of the intervals makes it seem like it might have been run on the whole genome, without that filtering.

Author: @jashapiro are you asking about the panel and WXS intervals or WGS? The first sentence is now about panel/WXS, which were derived from probes used to create the assays (these are always from a manufacturer of the kit). We also provided these in the data release. We can't link to them unless we link to the S3 data (the TGEN panel BED was from an earlier manufacturer and we had to get permission to use, so it is not public, except for our release). The second one states the BROAD intervals were used for WGS. Let me know what you think.

Reviewer: I think I was confused by the repeated reference to hg38 here. Perhaps just: 
```suggestion
Strelka2 [@doi:10/gdwrp4] v2.9.3 was run using default parameters on canonical chromosomes only (chr1-22, X,Y,M), as recommended by the authors.
```

Author: how about `Strelka2 [@doi:10/gdwrp4] v2.9.3 was run using default parameters for canonical chromosomes (chr1-22, X,Y,M) of human genome reference hg38, as recommended by the authors.` so that we have hg38 noted?

Reviewer: Everything is hg38, right? Its not specific to strelka...

Author: yes, correct... looks like we have inconsistencies - listing it many places

Author: removed since we have it in the alignment section

The final Strelka2 VCF was filtered for PASS variants.
Mutect2 from GATK v4.1.1.0 was run following Broad best practices outlined from their Workflow Description Language (WDL) [@url:https://github.com/broadinstitute/gatk/blob/4.1.1.0/scripts/mutect2_wdl/mutect2.wdl].
The final Mutect2 VCF was filtered for PASS variants. 

> To manage memory issues, VarDictJava [@doi:10.1093/nar/gkw227] v1.58 [@url:https://github.com/AstraZeneca-NGS/VarDictJava] was run using 20K interval chunks of the input BED, padded by 100 bp on each side, such that if an INDEL occurred in between intervals, it would be captured.

Reviewer: Are they chunks of 20K intervals, or 20Kb chunks?

Author: Kb, will fix

> Parameters and filtering followed BCBIO standards except that variants with a variant allele frequency (VAF) >= 0.05 (instead of >= 0.10) were retained. 

Reviewer: Link to BCBio?

Author: updated in upcoming commit

The 0.05 VAF increased the true positive rate for INDELs and decreased the false positive rate for SNVs when using VarDict in consensus calling. 
The final VCF was filtered for PASS variants with TYPE=StronglySomatic.
Lancet [@doi:10.1038/s42003-018-0023-9] v1.0.7 [@url:https://github.com/nygenome/lancet] was run using default parameters, unless noted below.
For input intervals to Lancet, a reference BED was created by using only the UTR, exome, and start/stop codon features of the GENCODE 31 reference and padded by 300 bp on each side. 

> Per recommendations by the New York Genome Center [@doi:10.1101/623702], the Lancet input intervals were augmented with PASS variant calls from Strelka2 and Mutect2 as validation.

Reviewer: ```suggestion
Per recommendations by the New York Genome Center [@doi:10.1101/623702], the Lancet input intervals were augmented with PASS variant calls from Strelka2 and Mutect2 (also padded by 300 bp on each side) as validation.
```

> Per recommendations by the New York Genome Center [@doi:10.1101/623702v1], the Lancet input intervals were augmented with PASS variant calls from Strelka2 and Mutect2 as validation.

Reviewer: ```suggestion
Per recommendations by the New York Genome Center [@doi:10.1101/623702v1], the Lancet input intervals were augmented with PASS variant calls from Strelka2 and Mutect2 (also padded by 300 bp on each side) as validation.
```

Author: No, the strelka2 and mutect2 were not padded at all (that's why I wrote non-padded above, unless otherwise noted below - to only specify within lancet and vardict, the padded.

Author: v2 removed in upcoming commit

Reviewer: This is a wording question.. I was interpreting this sentence to mean that the lancet intervals included everything in the previous sentence, plus 600 bp around every SNV called by strelka and mutect.

Author: ahh, hmm

Author: `Per recommendations by the New York Genome Center [@doi:10.1101/623702], the Lancet input intervals described above were augmented with PASS variant calls from Strelka2 and Mutect2 as validation.`?

> Per recommendations by the New York Genome Center [@doi:10.1101/623702v1], the Lancet input intervals were augmented with PASS variant calls from Strelka2 and Mutect2 as validation.

Reviewer: ```suggestion
Per recommendations by the New York Genome Center [@doi:10.1101/623702], the Lancet input intervals were augmented with PASS variant calls from Strelka2 and Mutect2 as validation.
```

> Per recommendations by the New York Genome Center [@doi:10.1101/623702], the Lancet input intervals described above were augmented with PASS variant calls from Strelka2 and Mutect2 as validation.

Reviewer: Maybe something like?
```suggestion
Per recommendations by the New York Genome Center [@doi:10.1101/623702], the Lancet input intervals described above were augmented with padded regions surrounding PASS variant calls from Strelka2 and Mutect2 as validation.
```

Author: I feel like this sounds like the variant calls from strelka/mutect were padded, but it was that the same files as above were augmented with strelka2/mutect2 calls, if not present already...

Reviewer: My understanding was that the variant calls *are* padded... That is if there was a call from strelka or mutect at base 500, the region 200-800 would be added to the lancet intervals (if it wasn't already in there).

Author: hmm, let me get clarification from @migbro - wasn't my understanding but I didn't run it :)

Author: ok... so I got some clarification: 
`for lancet, incoming regions consist of the CDS enhanced with variant calls not in the CDS from strelka2 and mutect2. just like in vardict, they will be padded by the software by 300bp, UNLIKE vardict, since it’s not really WGS, probably not a whole lot of overlap between intervals, so, yeah, in going throught this though exercise, the total bp length can vary a lot sample-to-sample` - i think we should reword as this is run with padding, input intervals not padded.

Author: and perhaps, TMB should only be calculated using mutect2/strelka bc they are not padded

Reviewer: > and perhaps, TMB should only be calculated using mutect2/strelka bc they are not padded

Agreed, though I think the main concern would be is less about padding than about the fact that every sample has its own input interval.

How about something like this for the lines in question:

```
For input intervals to Lancet, a reference BED was created by using only the UTR, exome, and start/stop codon features of the GENCODE 31 reference, augmented as recommended with PASS variant calls from Strelka2 and Mutect2 [@doi:10.1101/623702].
These intervals were then padded by 300 bp on each side during Lancet variant calling.

Author: right, yes that is what i was getting at. sounds better, will update.
